src/common.mak

```diff
@@ -0,0 +1,204 @@
+PHOBOS_BUILD_DIR=$(PHOBOS_GENERATED)/$(BUILD_TRIPLE)
+
+################################################################################
+# Detect build mode (default: release)
+################################################################################
+
+# Default to a release built, override with BUILD=debug
+ifeq (,$(BUILD))
+BUILD=release
+endif
+
+ifneq ($(BUILD),release)
+    ifneq ($(BUILD),debug)
+        $(error Unrecognized BUILD=$(BUILD), must be 'debug' or 'release')
+    endif
+    ENABLE_DEBUG := 1
+endif
+
+################################################################################
+# Target-specific flags
+################################################################################
+
+# default to PIC on x86_64, use PIC=1/0 to en-/disable PIC.
+# Note that shared libraries and C files are always compiled with PIC.
+ifeq ($(PIC),)
+    ifeq ($(MODEL),64) # x86_64
+        PIC:=1
+    else
+        PIC:=0
+    endif
+endif
+ifeq ($(PIC),1)
+    override PIC:=-fPIC
+else
+    override PIC:=
+endif
+
+
+ifeq (osx,$(OS))
+    export MACOSX_DEPLOYMENT_TARGET=10.7
+endif
+
+################################################################################
+# Define library files and import paths
+################################################################################
+
+DRUNTIME_IMPORT_DIR=$(DRUNTIME_DIR)/import
+PHOBOS_IMPORT_DIR=$(PHOBOS_DIR)
+
+ifneq (,$(DRUNTIME))
+	CUSTOM_DRUNTIME=1
+endif
+
+ifeq (,$(findstring win,$(OS)))
+   LIB_DRUNTIME = $(DRUNTIME_BUILD_DIR)/libdruntime.a
+   LIB_DRUNTIMESO = $(basename $(LIB_DRUNTIME)).so.a
+else
+   LIB_DRUNTIME = $(DRUNTIME_PATH)/lib/druntime.lib
+endif
+
+# Set PHOBOS name and full path
+ifeq (,$(findstring win,$(OS)))
+	LIB_PHOBOS = $(PHOBOS_BUILD_DIR)/libphobos2.a
+	LIB_PHOBOSSO = $(PHOBOS_BUILD_DIR)/libphobos2.so
+endif
+
+# build with shared library support
+# (defaults to true on supported platforms, can be overridden w/ make SHARED=0)
+SHARED=$(if $(findstring $(OS),linux freebsd),1,)
+LINKDL=$(if $(findstring $(OS),linux),-L-ldl,)
+
+################################################################################
+# Set CC and DMD
+################################################################################
+
+ifeq ($(OS),win32wine)
+	CC = wine dmc.exe
+	DMD = wine dmd.exe
+	RUN = wine
+else
+	DMD = $(DMD_BUILD_DIR)/dmd
+	ifeq ($(OS),win32)
+		CC = dmc
+	else
+		CC = cc
+	endif
+	RUN =
+endif
+
+################################################################################
+# Commonly used binaries
+################################################################################
+
+DUB=dub
+
+################################################################################
+# Platform-specific variables
+################################################################################
+
+# Set DOTOBJ and DOTEXE
+ifeq (,$(findstring win,$(OS)))
+	DOTOBJ:=.o
+	DOTEXE:=
+	PATHSEP:=/
+else
+	DOTOBJ:=.obj
+	DOTEXE:=.exe
+	PATHSEP:=$(shell echo "\\")
```

Comment:
Isn't there a simple means?

Reply:
This was copied from Phobos: https://github.com/dlang/phobos/blob/master/posix.mak#L160

Reply:
This works without invoking a subprocess:
```
PATHSEP := $(subst /,/,\)
```